fixtures/fun-generation/led-pot-12x1w-qcl-rgb-ww-40.json

```diff
@@ -0,0 +1,507 @@
+          ],
+          "type": "Effect",
+          "effectName": "Program 05",
+          "switchChannels": {
+            "Color Presets / Sound Control": "Color Presets"
+          }
+        },
+        {
+          "dmxRange": [
+            85,
+            101
+          ],
+          "type": "Effect",
+          "effectName": "Program 06",
+          "switchChannels": {
+            "Color Presets / Sound Control": "Color Presets"
+          }
+        },
+        {
+          "dmxRange": [
+            102,
+            118
+          ],
+          "type": "Effect",
+          "effectName": "Program 07",
+          "switchChannels": {
+            "Color Presets / Sound Control": "Color Presets"
+          }
+        },
+        {
+          "dmxRange": [
+            119,
+            135
+          ],
+          "type": "Effect",
+          "effectName": "Program 08",
+          "switchChannels": {
+            "Color Presets / Sound Control": "Color Presets"
+          }
+        },
+        {
+          "dmxRange": [
+            136,
+            152
+          ],
+          "type": "Effect",
+          "effectName": "Program 09",
+          "switchChannels": {
+            "Color Presets / Sound Control": "Color Presets"
+          }
+        },
+        {
+          "dmxRange": [
+            153,
+            169
+          ],
+          "type": "Effect",
+          "effectName": "Program 10",
+          "switchChannels": {
+            "Color Presets / Sound Control": "Color Presets"
+          }
+        },
+        {
+          "dmxRange": [
+            170,
+            186
+          ],
+          "type": "Effect",
+          "effectName": "Program 11",
+          "switchChannels": {
+            "Color Presets / Sound Control": "Color Presets"
+          }
+        },
+        {
+          "dmxRange": [
+            187,
+            203
+          ],
+          "type": "Effect",
+          "effectName": "Program 12",
+          "switchChannels": {
+            "Color Presets / Sound Control": "Color Presets"
+          }
+        },
+        {
+          "dmxRange": [
+            204,
+            220
+          ],
+          "type": "Effect",
+          "effectName": "Program 13",
+          "switchChannels": {
+            "Color Presets / Sound Control": "Color Presets"
+          }
+        },
+        {
+          "dmxRange": [
+            221,
+            237
+          ],
+          "type": "Effect",
+          "effectName": "Program 14",
+          "switchChannels": {
+            "Color Presets / Sound Control": "Color Presets"
+          }
+        },
+        {
+          "dmxRange": [
+            238,
+            255
+          ],
+          "type": "Effect",
+          "effectName": "Sound control",
+          "soundControlled": true,
+          "switchChannels": {
+            "Color Presets / Sound Control": "Sound Control"
+          }
+        }
+      ]
+    },
+    "Color Presets": {
```

Comment:
The "Colour macros programme 01" values all say "if channel 6 = 1 … 16".  Do we need a third switch, like "Color Presets / Sound Control / Standard", or is it obvious that this channel is dead when not at "Program 01" or "Sound Control"?

Reply:
The other fun generation fixtures are implemented the same, so this should be fine IMO.

Reply:
Well, I don't know if this is such a case, but OFL has made mistakes before.  :-)

I'm coming at this from the perspective of writing new software to use OFL fixture data, so I don't know how existing lighting software works.  Consider the case:
 - the Program channel is set to 0 (the default value)
 - then I set the Color Presets channel to 32 (green)

There's no way that I see for the software to know that this won't work (without a channel switch to encode the information that "Color Presets depends on the Program being set to 01").  Is that something users expect to work automatically?

Reply:
That makes sense!

I added a `No Function` switching channel, which is probably more descriptive than `Standard`. WDYT?

Reply:
@kengruven

Reply:
Looks good to me.